packages/workspace/src/generators/new/new.ts

```diff
@@ -130,7 +130,12 @@ async function initializeGitRepo(
     );
     return;
   }
-  await execute(['init']);
+  if (gte(gitVersion, '2.28.0')) {
+    await execute(['init', '-b', deduceDefaultBase()]);
```

Comment:
Shouldn't `options.defaultBase` take precedent over the default base?

Also.. shouldn't `git init` be able to find deduce the default base itself?

I believe the logic should be the following:

1. If no `defaultBase` is passed here, run `git init`
2. If a `defaultBase` is passed here, run `git init -b ${options.defaultBase}`

Reply:
Yeah, thats true. It was definitely supposed to. I'll get that updated.

Reply:
So if a user has not configured git to create with `main`, git may initialize with master. I seen this with git v2.33, I'm not sure if it will change.

By using the deduceDefaultBase logic in both places (here, and nx.json), we ensure that there is never a difference caused on our side (and the user's experience in default matches the docs).